From: Daniel Stenberg <daniel@haxx.se>
Date: Wed, 6 Apr 2011 10:03:11 +0000 (+0200)
Subject: configure: fix $VERSION
X-Git-Url: http://git.libssh2.org/?p=libssh2.git;a=commitdiff_plain;h=f0a37bdadc311b83fcb2ccafad4827016abe5d18;js=1

configure: fix $VERSION

Stop using the $VERSION variable as it seems to be magically used by
autoconfig itself and thus gets set to the value set in AC_INIT()
without us wanting that. $LIBSSH2VER is now the libssh2 version as
detected.

Reported by: Paul Howarth
Bug: http://www.libssh2.org/mail/libssh2-devel-archive-2011-04/0008.shtml
---

diff --git a/configure.ac b/configure.ac
index 9bdf3b9..46c6cfa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -15,12 +15,14 @@ if test "x$SED" = "xsed-was-not-found-by-configure"; then
 fi
 
 dnl figure out the libssh2 version
-VERSION=`$SED -ne 's/^#define LIBSSH2_VERSION *"\(.*\)"/\1/p' ${srcdir}/include/libssh2.h`
+LIBSSH2VER=`$SED -ne 's/^#define LIBSSH2_VERSION *"\(.*\)"/\1/p' ${srcdir}/include/libssh2.h`
 AM_INIT_AUTOMAKE
 AC_MSG_CHECKING([libssh2 version])
-AC_MSG_RESULT($VERSION)
+AC_MSG_RESULT($LIBSSH2VER)
 
-AB_VERSION=$VERSION
+AC_SUBST(LIBSSH2VER)
+
+AB_VERSION=$LIBSSH2VER
 
 AB_INIT
 
@@ -289,7 +291,7 @@ AC_OUTPUT
 
 AC_MSG_NOTICE([summary of build options:
 
-  version:          ${VERSION}
+  version:          ${LIBSSH2VER}
   Host type:        ${host}
   Install prefix:   ${prefix}
   Compiler:         ${CC}
diff --git a/libssh2.pc.in b/libssh2.pc.in
index 013e6bf..eaef88b 100644
--- a/libssh2.pc.in
+++ b/libssh2.pc.in
@@ -10,7 +10,7 @@ includedir=@includedir@
 Name: libssh2
 URL: http://www.libssh2.org/
 Description: Library for SSH based connunication
-Version: @VERSION@
+Version: @LIBSSH2VER@
 Libs: -L${libdir} -lssh2 @LDFLAGS@ @LIBS@
 Libs.private: @LIBS@
 Cflags: -I${includedir}
